src/librustc_trans/back/msvc/mod.rs

```diff
@@ -31,8 +31,18 @@
 //! paths/files is based on Microsoft's logic in their vcvars bat files, but
 //! comments can also be found below leading through the various code paths.
 
+// A simple macro to make this option mess easier to read
+macro_rules! otry {
+    ($expr:expr) => (match $expr {
+        Some(val) => val,
+        None => return None,
+    })
+}
```

Comment:
This macro has been written at least twice in `librustc` under different names 😆 

* `try_opt!`: https://github.com/rust-lang/rust/blob/382ab92ceedc258e794c1a95aef21d3be3fa76c4/src/librustc/dep_graph/dep_node.rs#L13-L20
* `option_try!`: https://github.com/rust-lang/rust/blob/bb4b3fb7f97924919f072ec9a360bdf943218dbf/src/librustc/util/common.rs#L72-L74